packages/common/src/store/purchase-content/sagas.ts

```diff
@@ -207,21 +220,38 @@ function* doStartPurchaseContentFlow({
         .div(BN_USDC_CENT_WEI)
         .toNumber()
       yield* put(buyUSDC())
-      yield* put(
-        onrampOpened({
-          provider: USDCOnRampProvider.STRIPE,
-          purchaseInfo: {
-            desiredAmount: balanceNeededCents
-          }
+      let result: { succeeded?: any; failed?: any; canceled?: any } | null =
+        null
+      if (isBuyUSDCViaSolEnabled) {
+        yield* put(
+          buyCryptoViaSol({
+            // expects "friendly" amount, so dollars
+            amount: balanceNeededCents / 100.0,
+            mint: 'usdc',
+            provider: OnRampProvider.STRIPE
+          })
+        )
+        result = yield* race({
+          succeeded: take(buyCryptoSucceeded),
+          failed: take(buyCryptoFailed),
+          canceled: take(buyCryptoCanceled)
         })
-      )
-
-      const result = yield* race({
-        success: take(buyUSDCFlowSucceeded),
-        canceled: take(onrampCanceled),
-        failed: take(buyUSDCFlowFailed)
-      })
+      } else {
+        yield* put(
+          onrampOpened({
+            provider: USDCOnRampProvider.STRIPE,
+            purchaseInfo: {
+              desiredAmount: balanceNeededCents
```

Comment:
Oh wait, so in the buy-crypto flow we have to pass it in dollars, but in this flow it's in cents?